client/pages/edit-event.jsx

```diff
@@ -0,0 +1,172 @@
+import React from 'react';
+import DatePicker from 'react-datepicker';
+
+export default class EditEvent extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      event: {
+        title: '',
+        date: '',
+        startingtime: '',
+        address: '',
+        city: '',
+        state: '',
+        mapLocation: { }
+      },
+      showModal: false
+    };
+    this.handleUpdate = this.handleUpdate.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.handleDate = this.handleDate.bind(this);
+    this.handleAddress = this.handleAddress.bind(this);
+    this.handleCity = this.handleCity.bind(this);
+    this.handleState = this.handleState.bind(this);
+  }
+
+  componentDidMount() {
+    fetch(`/api/events/${this.props.eventID}`)
+      .then(res => res.json())
+      .then(data => this.setState({ event: data }));
```

Comment:
Your `fetch` promises should include a `.catch` handler that notifies the user if something goes wrong (for example, their internet may be temporarily disconnected).